core/unit_test/TestPageMigration.hpp

```diff
@@ -0,0 +1,242 @@
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#if defined _WIN32  // windows system
+#include <windows.h>
+unsigned getBytesPerPage() {
+  SYSTEM_INFO si;
+  GetSystemInfo(&si);
+  return si.dwPageSize;
+}
+#else  // unix/posix system
+#include <unistd.h>
+unsigned getBytesPerPage() { return sysconf(_SC_PAGESIZE); }
+#endif
+
+#include <Kokkos_Core.hpp>
+
+#include <algorithm>
+#include <numeric>
+#include <iostream>
+
+namespace {
+void printTimings(std::ostream& out, std::vector<double> const& tr) {
+  out << "TimingResult contains " << tr.size() << " results:\n";
+  for (auto it = tr.begin(); it != tr.end(); ++it) {
+    out << "Duration of loop " << it - tr.begin() << " is " << *it
+        << " seconds\n";
+  }
+}
+
+template <typename T>
+T computeMean(std::vector<T> const& results) {
+  return std::accumulate(results.begin(), results.end(), T{}) / results.size();
+}
+
+template <typename ExecSpace, typename ViewType>
+std::vector<double> incrementInLoop(ViewType& view,
+                                    unsigned int numRepetitions) {
+  Kokkos::Timer timer;
+  std::vector<double> results;
+
+  Kokkos::fence();
+  for (unsigned i = 0; i < numRepetitions; ++i) {
+    timer.reset();
+    Kokkos::parallel_for(
+        "increment",
+        Kokkos::RangePolicy<ExecSpace,
+                            Kokkos::IndexType<decltype(view.size())>>{
+            0, view.size()},
+        KOKKOS_LAMBDA(auto i) { ++view(i); });
+    Kokkos::fence();
+    results.push_back(timer.seconds());
+  }
+  return results;
+}
+
+size_t getDeviceMemorySize() {
+#if defined KOKKOS_ENABLE_CUDA
+  return Kokkos::Cuda{}.cuda_device_prop().totalGlobalMem;
+#elif defined KOKKOS_ENABLE_HIP
+  return Kokkos::Experimental::HIP::hip_device_prop().totalGlobalMem;
+#elif defined KOKKOS_ENABLE_SYCL
+  return 1024 * 1024 * 1024 *
+         1024;  // 1gb as I can not access sycl::info::device via Kokkos
```

Comment:
`Kokkos::Experimental::SYCL{}.sycl_queue()` gives use the `sycl::queue` member which has a `get_device()` member function returning a `sycl::device` object that has a `get_info()` member function that allows for querying `info::device::
max_mem_alloc_size` or `info::device::global_mem_size`.

Reply:
done in [2f1bf3](https://github.com/kokkos/kokkos/pull/5289/commits/2f1bf31480c12c8df25ddbf3405bbcb016fd1b10)